sources/web/datalab/static/datalab.js

```diff
@@ -277,6 +277,166 @@ function initializePage(dialog, saveFn) {
   });
 }
 
+// constants for minitoolbar operations
+const CELL_METADATA_COLLAPSED = 'hiddenCell';
+const CELL_METADATA_OUTPUT_COLLAPSED = 'hiddenOutput';
+const COLLAPSE_BUTTON_CLASS = 'fa-eye';
+const UNCOLLAPSE_BUTTON_CLASS = 'fa-eye-slash';
+const COLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-compress';
+const UNCOLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-expand';
+const CLEAR_BUTTON_CLASS = 'fa-minus-square-o';
+const RUN_BUTTON_CLASS = 'fa-play';
+const COLLAPSED_CELL_INNER_HTML = '-- Collapsed cell --';
+
+/**
+ * Collapse entire cell
+ */
+function toggleCollapseCell(cell, hide) {
+  // the cell is currently hidden if its input div is
+  let inputDiv = cell.cell_type === 'code' ? 'div.input' : 'div.inner_cell';
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.element.find(inputDiv)[0].style.display === 'none';
+  collapseSpan = cell.element.find('span.collapse-cell')[0];
+  if (isHidden) {
+    cell.element.find(inputDiv).show();
+    // for code cells, also show the output div
+    if (cell.cell_type == 'code')
+      cell.element.find('div.output_wrapper').show();
+    cell.element.find('div.cellPlaceholder').hide();
+    collapseSpan.classList.add(UNCOLLAPSE_BUTTON_CLASS);
+    collapseSpan.classList.remove(COLLAPSE_BUTTON_CLASS);
+  } else {
+    cell.element.find(inputDiv).hide();
+    // for code cells, also hide the output div
```

Comment:
I think there is a compelling use case for showing the output but not the input (e.g. for dashboards where the input is an implementation detail and what the user really cares about is the output).

As such, I'd prefer if the collapsing of the code and output could be separated.